### PR TITLE
2807 new module svtyper

### PR DESCRIPTION
**Pull request by kubranarci**
## PR checklist

Closes #XXX 

- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by nvnieuwk** [comment]
@nf-core-bot fix linting please

File: modules/nf-core/svtyper/svtyper/main.nf
Comment (nvnieuwk) [inline modules/nf-core/svtyper/svtyper/main.nf:34]:
```suggestion
        --output_vcf ${prefix}.vcf \\
```
Don't use custom prefixes aside from the file extension :) The pipeline developers should add this themselves through the prefix (or not if they don't want to)

Comment (nvnieuwk) [reply modules/nf-core/svtyper/svtyper/main.nf:34]: You can also add a check for file name collisions as shown [here](https://nf-co.re/docs/contributing/modules#naming-conventions) part 6

File: modules/nf-core/svtyper/svtyper/main.nf
Comment (nvnieuwk) [inline modules/nf-core/svtyper/svtyper/main.nf:36]:
```suggestion
        --write_alignment ${prefix}.bam \\
```

Same here :)

File: modules/nf-core/svtyper/svtyper/main.nf
Comment (nvnieuwk) [inline modules/nf-core/svtyper/svtyper/main.nf:49]:
```suggestion
    touch ${prefix}.vcf
    touch ${prefix}.bam
```

File: modules/nf-core/svtyper/svtyper/meta.yml
Comment (nvnieuwk) [inline modules/nf-core/svtyper/svtyper/meta.yml:22]: Can you also add meta3 here?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/nf-core/svtyper/svtyper/main.nf
Comment (kubranarci) [reply modules/nf-core/svtyper/svtyper/main.nf:34]: gt.vcf was the way that developers use. Besides, what if input and output names became the same at the end? How will the output be differentiated? That did not make sense to be

Comment (nvnieuwk) [reply modules/nf-core/svtyper/svtyper/main.nf:34]: You can read the link I sent you in my previous message, this explains everything. You can still add the `.gt` yourself through prefix if you want to use it :)

**Comment by kubranarci** [comment]
@nf-core-bot fix linting pretty please :pray:

**Comment by nvnieuwk** [comment]
> I went through a lot of new modules and I saw a lot of examples of adding extensions there. To be honest, l cannot find like any similar one you requested from me? Here, output files specific to the tool itself. I don't think naming would be an issue. Besides, It would make it hard to grep the correct output here.

This is a pretty recent guideline so not all modules are up to date yet. You should add a check that fails the process when both the input and the output files are called the same. I get why you are saying this, but by removing fixed suffices, we ensure the most flexibility for pipeline developers using this module, which is the most important thing for nf-core modules

**Comment by kubranarci** [comment]
> > I went through a lot of new modules and I saw a lot of examples of adding extensions there. To be honest, l cannot find like any similar one you requested from me? Here, output files specific to the tool itself. I don't think naming would be an issue. Besides, It would make it hard to grep the correct output here.
> 
> This is a pretty recent guideline so not all modules are up to date yet. You should add a check that fails the process when both the input and the output files are called the same. I get why you are saying this, but by removing fixed suffices, we ensure the most flexibility for pipeline developers using this module, which is the most important thing for nf-core modules

The way you suggest prevents the usage of different prefixes for output types and makes it even more complicated. Seriously, even for recent pushes (3 hr ago) additional suffixes are there :D

**Comment by nvnieuwk** [comment]
> > > I went through a lot of new modules and I saw a lot of examples of adding extensions there. To be honest, l cannot find like any similar one you requested from me? Here, output files specific to the tool itself. I don't think naming would be an issue. Besides, It would make it hard to grep the correct output here.
> > 
> > 
> > This is a pretty recent guideline so not all modules are up to date yet. You should add a check that fails the process when both the input and the output files are called the same. I get why you are saying this, but by removing fixed suffices, we ensure the most flexibility for pipeline developers using this module, which is the most important thing for nf-core modules
> 
> The way you suggest prevents the usage of different prefixes for output types and makes it even more complicated. Seriously, even for recent pushes (3 hr ago) additional suffixes are there :D

You can change output file names when publishing them to the output directory. But I get what you mean, I'm just saying what the guidelines are for this. @nf-core/maintainers can I get some more opinions on this?

File: modules/nf-core/svtyper/svtyper/main.nf
Comment (maxulysse) [inline modules/nf-core/svtyper/svtyper/main.nf:20]:
```suggestion
    tuple val(meta), path("*.json"), emit: json
    tuple val(meta), path("*.vcf") , emit: gt_vcf
    tuple val(meta), path("*.bam") , emit: bam
    path "versions.yml"            , emit: versions
```

**Comment by SPPearce** [comment]
The failure [https://github.com/nf-core/modules/actions/runs/4573550127/jobs/8074167388?pr=3225#step:12:54](is the md5sum for the bam file).
It is very common for a bam file to be different between docker/singularity and conda. The bam file contains paths in its header, and they end up slightly different on conda. 
Two options:

1. Don't test the md5sum for the bam file, if it isn't the main output of the tool.
2. Ignore the failing conda test and merge anyway.

**Comment by SPPearce** [comment]
It looks like the bam file is not the most important part of this module, so I would be inclined to swap to just checking that the bam file exists, rather than its md5sum.

**Comment by kubranarci** [comment]
> The failure [is on the md5sum for the bam file.](https://github.com/nf-core/modules/actions/runs/4573550127/jobs/8074167388?pr=3225#step:12:54) It is very common for a bam file to be different between docker/singularity and conda. The bam file contains paths in its header, and they end up slightly different on conda. Two options:
> 
> 1. Don't test the md5sum for the bam file, if it isn't the main output of the tool.
> 2. Ignore the failing conda test and merge anyway.

Thank you! Then I will be waiting for reviewers' approval to merge this.

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Pull request merged**
